modules/core/src/main/scala/ch/epfl/bluebrain/nexus/commons/search/Pagination.scala

```diff
@@ -1,9 +1,39 @@
 package ch.epfl.bluebrain.nexus.commons.search
 
+import io.circe.Json
+
+/**
+  * Base request pagination data type.
+  *
+  */
+trait Pagination {
+
+  /**
+    *
+    * @return size the maximum number of results per page
+    */
+  def size: Int
+}
+
 /**
-  * Request pagination data type.
+  * Request pagination data type using `from`.
   *
   * @param from the start offset
   * @param size the maximum number of results per page
   */
-final case class Pagination(from: Long, size: Int)
+final case class FromPagination(from: Int, size: Int) extends Pagination
+
+/**
+  * Request pagination data type using `search_after`.
+  *
+  * @param searchAfter  [[Json]] array to use as ElasticSearch `search_after` field.
+  * @param size         the maximum number of results per page
+  */
+final case class SearchAfterPagination(searchAfter: Seq[Json], size: Int) extends Pagination
```

Comment:
Is there a reason why you want here a free form Json rather than a data type which represents the search after? 
Something like this for example:
```scala
final case class SearchAfter(offset: Long, key: String)
```

Reply:
It has to be free form Json, because `search_after`  depends on the `sort` used in the ElasticSearch query.

Reply:
You can use an `HList` where every element implements `Encoder[A]` adn build the Json from there.

but yeah, if you don't want to make it complicated, it is fine how it is